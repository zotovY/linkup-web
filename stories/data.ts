export const user: User = {
    name: "Yaroslav Zotov",
    id: 1,
    email: "the1ime@yandex.ru",
    createdAt: new Date(),
    password: "12345678",
    theme: 0,
    links: [],
    profileImagePath: "https://sun9-70.userapi.com/impg/Afr3SX4BeAJzLemVyNMhHhr-5qvMlj1MedV0wA/5B6s0MBC6Y8.jpg?size=1728x2160&quality=96&sign=7a48ae45b2c91959995e58331fa10926&type=album"
}

export const link: Link = {
    createdAt: new Date(),
    href: "https://github.com/zotovY",
    iconName: "logo-github",
    id: 1,
    subtitle: "This is my github account",
    title: "Github",
    user: user,
}
