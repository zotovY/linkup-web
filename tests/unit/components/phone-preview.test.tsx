/// <reference types="@types/jest" />
import { render } from "../../test-utils";
import '@testing-library/jest-dom';
import 'jest-styled-components';

import PhonePreview, { Props } from "@/components/phone-preview";
import { cleanup } from "@testing-library/react";
import { user } from "../../data";

describe("Testing PhonePreview component", () => {

    afterEach(cleanup);

    const props: Props = user;

    test("should renders correctly", () => {
        const { asFragment } = render(<PhonePreview { ...props } />);
        expect(asFragment()).toMatchSnapshot();
    });
});
