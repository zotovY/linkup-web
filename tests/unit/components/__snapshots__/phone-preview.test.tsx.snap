// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PhonePreview component should renders correctly 1`] = `
<DocumentFragment>
  .c2 {
  width: 90px;
  height: 90px;
  border-radius: 50%;
  background-size: cover;
  background-position: center;
  background-repeat: no-repeat;
  background-color: #D9DDE9;
}

.c3 {
  font-size: 20px;
  font-weight: 500;
  text-align: center;
  margin-top: 11px;
  color: #14142b;
}

.c1 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c8 {
  position: absolute;
  bottom: 15px;
  left: 0;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 18px;
  font-weight: 500;
  color: #5f2eea;
  cursor: pointer;
}

.c8 img {
  margin-right: 10px;
}

.c6 svg {
  height: 36px;
  width: 36px;
  color: #5f2eea;
  fill: #5f2eea;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 10px;
  border-radius: 16px;
  margin-bottom: 20px;
  background-color: #FFFFFF;
  -webkit-transition: background-color 200ms ease;
  transition: background-color 200ms ease;
  cursor: pointer;
}

.c5:hover {
  background-color: #fEfEfE;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: calc(100% - 46px);
}

.c7 span.title {
  font-size: 15px;
  font-weight: 500;
  color: #14142b;
}

.c7 span.subtitle {
  font-size: 12px;
  color: #4e4b66;
}

.c0 {
  width: 350px;
  height: 740px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  border-radius: 50px;
  box-shadow: 0 0 0 15px #111111;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #f7f7fc;
  padding: 0 20px;
  position: relative;
  overflow: hidden;
}

.c0 .UserInformationPreview {
  margin-top: 30px;
}

.c4 {
  margin-top: 51px;
  width: 100%;
}

<div
    class="c0 phone-preview"
  >
    <div
      class="c1 UserInformationPreview"
    >
      <div
        class="c2 user-avatar"
        data-testid="avatar"
        style="background-image: url(https://sun9-70.userapi.com/impg/Afr3SX4BeAJzLemVyNMhHhr-5qvMlj1MedV0wA/5B6s0MBC6Y8.jpg?size=1728x2160&quality=96&sign=7a48ae45b2c91959995e58331fa10926&type=album);"
      />
      <p
        class="c3"
      >
        Yaroslav Zotov
      </p>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <span
          class="c6"
          color="#5f2eea"
          height="36px"
          rotate="0"
          width="36px"
        >
          <svg
            class=""
            style="margin-right: 10px; margin-top: 3px;"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 32C132.3 32 32 134.9 32 261.7c0 101.5 64.2 187.5 153.2 217.9a17.56 17.56 0 003.8.4c8.3 0 11.5-6.1 11.5-11.4 0-5.5-.2-19.9-.3-39.1a102.4 102.4 0 01-22.6 2.7c-43.1 0-52.9-33.5-52.9-33.5-10.2-26.5-24.9-33.6-24.9-33.6-19.5-13.7-.1-14.1 1.4-14.1h.1c22.5 2 34.3 23.8 34.3 23.8 11.2 19.6 26.2 25.1 39.6 25.1a63 63 0 0025.6-6c2-14.8 7.8-24.9 14.2-30.7-49.7-5.8-102-25.5-102-113.5 0-25.1 8.7-45.6 23-61.6-2.3-5.8-10-29.2 2.2-60.8a18.64 18.64 0 015-.5c8.1 0 26.4 3.1 56.6 24.1a208.21 208.21 0 01112.2 0c30.2-21 48.5-24.1 56.6-24.1a18.64 18.64 0 015 .5c12.2 31.6 4.5 55 2.2 60.8 14.3 16.1 23 36.6 23 61.6 0 88.2-52.4 107.6-102.3 113.3 8 7.1 15.2 21.1 15.2 42.5 0 30.7-.3 55.5-.3 63 0 5.4 3.1 11.5 11.4 11.5a19.35 19.35 0 004-.4C415.9 449.2 480 363.1 480 261.7 480 134.9 379.7 32 256 32z"
            />
          </svg>
        </span>
        <div
          class="c7"
        >
          <span
            class="title"
          >
            Github
          </span>
          <span
            class="subtitle"
          >
            This is my github account
          </span>
        </div>
      </div>
    </div>
    <div
      class="c8"
    >
      <img
        alt="linkup logo small"
        src="/images/logo-small.png"
      />
      <span>
        Linkup
      </span>
    </div>
  </div>
</DocumentFragment>
`;
